pantheon-bundle/src/main/java/com/redhat/pantheon/servlet/PublishDraftVersion.java

```diff
@@ -179,4 +179,14 @@ protected void doRun(SlingHttpServletRequest request, PostResponse response, Lis
             throw new RepositoryException(ex.getMessage());
         }
     }
+
+    @NotNull
+    private Optional<? extends Document> getDocument(SlingHttpServletRequest request, ResourceResolver serviceResourceResolver) {
+        if(serviceResourceResolver.getResource(request.getResource().getPath()).adaptTo(Document.class).getResourceType().equalsIgnoreCase("pantheon/assembly")){
+            Assembly assembly = serviceResourceResolver.getResource(request.getResource().getPath()).adaptTo(Assembly.class);
+            return Optional.ofNullable(assembly);
+        }
+            Module module = serviceResourceResolver.getResource(request.getResource().getPath()).adaptTo(Module.class);
+            return Optional.ofNullable(module);
+    }
```

Comment:
I have a bunch of thoughts about this method.

- It doesn't make a whole lot of sense to use both `@NotNull` and the `Optional` construct together. The whole point of Optional is to wrap something that may be null, so if your method returns an Optional, then of course the reference to the Optional itself is never going to be null (even if the thing that it wraps might be). Having said that, Sling makes sure that your servlet is only called when your Resource actually exists, so we don't need to worry about a null check. If I weren't about to recommend that you just remove this method, then I would recommend that you just drop both constructs because it's not worth worrying about.
- The `if ()` statement here is a bit long. If I were writing this method, here's how I'd do it instead:
```
Resource r = serviceResourceResolver.getResource(request.getResource().getPath());
if ("pantheon/assembly".equals(r.getResourceType()) {
    return r.adaptTo(Assembly.class);
} else {
    return r.adaptTo(Module.class);
}
```
- However, you really don't need the method at all. Just call `.adaptTo(Document.class)` on line 123 and use that for the entire servlet. Since you need to know what type of event to fire on line 97, just check the resource type there. The rest of the servlet can be completely generic.
- If you get rid of the unnecessary Optional construct, you don't need `.get()` cluttering up the code.